tests/conformance/state/state.go

```diff
@@ -298,6 +298,45 @@ func ConformanceTests(t *testing.T, props map[string]string, statestore state.St
 		})
 	}
 
+	if config.HasOperation("get") && config.HasOperation("set") {
+		t.Run("read, write, read output match", func(t *testing.T) {
+			for _, scenario := range scenarios {
+				if !scenario.bulkOnly && !scenario.transactionOnly {
+					t.Logf("Retrieving value for %s", scenario.key)
+					req := &state.GetRequest{
+						Key: scenario.key,
+					}
+					if len(scenario.contentType) != 0 {
+						req.Metadata = map[string]string{metadata.ContentType: scenario.contentType}
+					}
+					res, err := statestore.Get(req)
+					assert.Nil(t, err)
+					assertEquals(t, scenario.value, res)
+					t.Logf("Rewriting retrieved value for %s", scenario.key)
+					setReq := &state.SetRequest{
+						Key:   scenario.key,
+						Value: res,
```

Comment:
This is an issue, the initial `SetRequest` (line 241) uses `scenario.value` (proper Go type) as the `Value` (interface{}) field. But here the whole response is set as the `Value` field. That needs to change. 
Discounting that even if we set the `Value` field to `res.Data` it will not be the same as the previous set request, since the previous set request, the `scenario.value` variable was having proper Go type of the value and not `[]byte` which is the type for `res.Data`. So a proper type conversion needs to be done for this Set call to mimic the initial set so that it can be compared back to the `scenario.value` value. 


Again we need to use the type of `scenario.value` to unmarshal, `res.Data` properly before using that here. 
The below code snippet is essentially a copy of the `assertEquals` method defined at the end of file.
```go

        var d interface{}
	switch v := scenario.value.(type) {
	case intValueType:
		// Custom type requires case mapping
		if err := json.Unmarshal(res.Data, &v); err != nil {
			assert.Failf(t, "unmarshal error", "error: %w, json: %s", err, string(res.Data))
		}
		d = v
	case ValueType:
		// Custom type requires case mapping
		if err := json.Unmarshal(res.Data, &v); err != nil {
			assert.Failf(t, "unmarshal error", "error: %w, json: %s", err, string(res.Data))
		}
		d = v
	case int:
		// json.Unmarshal to float64 by default, case mapping to int coerces to int type
		if err := json.Unmarshal(res.Data, &v); err != nil {
			assert.Failf(t, "unmarshal error", "error: %w, json: %s", err, string(res.Data))
		}
		d = v
	case []byte:
		d = v
	default:
		// Other golang primitive types (string, bool ...)
		if err := json.Unmarshal(res.Data, &v); err != nil {
			assert.Failf(t, "unmarshal error", "error: %w, json: %s", err, string(res.Data))
		}
		d = v
	}
```

And use `d` as the `SetRequest.Value` field in line 318, since now `d` is properly typed and will mimic the `SetRequest` from line 241.

Reply:
Or was it the actual intention to store the response data as such (along with key, etag and value)?